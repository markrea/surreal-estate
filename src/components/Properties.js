import React, { useState, useEffect } from "react";
import {useLocation } from "react-router-dom";
import PropertyCard from "./PropertyCard";
import Alert from "./Alert";
import SideBar from "./SideBar";
import "../styles/Properties.css";

import axios from "axios";

const Properties = () => {
  const [properties, setProperties] = useState([]);
  const [alert, setAlert] = useState({ message: "", isSuccess: true });
  useEffect(() => {
    axios
      .get(`http://localhost:4000/api/v1/PropertyListing`)
      .then(({ data }) => setProperties(data))
      .catch((err) => {
        setAlert({
          message: "Unable to find properties.",
          isSuccess: false,
        });
      });
  }, []);

  const { search } = useLocation();
  useEffect(() => {
    axios.get(`http://localhost:4000/api/v1/PropertyListing${search}`)
    .then(({data}) => setProperties(data))
    .catch(err => console.log(err));
  }, [search]);

  return (
    <div className="property-body">
      <SideBar />
      {properties.map((property) => (
        <div key={property._id} className="col">
          <PropertyCard {...property} />
        </div>
      ))}
      <Alert message={alert.message} success={alert.isSuccess} />
    </div>
  );
};

export default Properties;
