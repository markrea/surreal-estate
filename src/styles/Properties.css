.property-body {
    padding: 0.5rem;
    
    
  }
  
  .col{
    display: inline-block;
    margin: auto;
    width: 90%;
    padding: 0.5rem;
    box-sizing: border-box;
  }

