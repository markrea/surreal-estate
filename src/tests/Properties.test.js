import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import Properties from "../components/Properties";

describe("Properties", () => {
  test("the page renders correctly against snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router history={history}>
        <Properties />
      </Router>
    );
    const component = asFragment();

    expect(component).toMatchSnapshot();
  });
});