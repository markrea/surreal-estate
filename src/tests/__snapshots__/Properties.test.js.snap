// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties the page renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div
    class="property-body"
  >
    <div>
      <a
        class="sidebar-link"
        href="/?query={\\"city\\":\\"Manchester\\"}"
      >
        Manchester
      </a>
      <a
        class="sidebar-link"
        href="/?query={\\"city\\":\\"Liverpool\\"}"
      >
        Liverpool
      </a>
      <a
        class="sidebar-link"
        href="/?query={\\"city\\":\\"Sheffield\\"}"
      >
        Sheffield
      </a>
      <a
        class="sidebar-link"
        href="/?query={\\"city\\":\\"Leeds\\"}"
      >
        Leeds
      </a>
    </div>
  </div>
</DocumentFragment>
`;
